File: Java-Kotlin/src/main/kotlin/string/Maximum Number of Balloons.kt
package string

/**
 * https://leetcode.com/problems/maximum-number-of-balloons/
 * */

fun maxNumberOfBalloons(text: String): Int {
    val count = IntArray(26)
    text.forEach { count[it - 'a']++ }
    count['l' - 'a'] /= 2
    count['o' - 'a'] /= 2
    val str = "balloon"
    var result = Int.MAX_VALUE
    str.forEach { result = Math.min(count[it - 'a'], result) }
    return result
}

File: C++/String/Maximum Number of Balloons.cpp
#include <bits/stdc++.h>

using namespace std;

/**
 * https://leetcode.com/problems/maximum-number-of-balloons/
 * */

class Solution {
public:
    string balloon = "balloon";
    int maxNumberOfBalloons(string text) {
        vector<int> count(26);
        for (char &ch:text) count[ch - 'a']++;
        count['l' - 'a'] /= 2, count['o' - 'a'] /= 2;
        int result = INT_MAX;
        for (char &ch:balloon) result = min(count[ch - 'a'], result);
        return result;
    }
};
